src/main/java/com/github/checkstyle/regression/module/ModuleUtils.java

```diff
@@ -0,0 +1,172 @@
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a checkstyle utility class.
+     */
+    public static boolean isCheckstyleUtility(GitChange change) {
+        final boolean returnValue;
+        if (isJavaMainSource(change)) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            returnValue = !NAME_TO_MODULE_EXTRAT_INFO.containsKey(fullName);
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a Java main source file.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a Java main source file.
+     */
+    private static boolean isJavaMainSource(GitChange change) {
+        final boolean returnValue;
+        final Matcher matcher = JAVA_SOURCE_PARTTEN.matcher(change.getPath());
+        if (matcher.find()) {
+            returnValue = "main".equals(matcher.group(1));
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Checks whether the corresponding file of a change may be considered as
+     * a test of checkstyle module.
+     * @param change change to check
+     * @return true if the corresponding file of a change may be considered as
+     *      a test of checkstyle module
+     */
+    public static boolean isCheckstyleModuleTest(GitChange change) {
+        final boolean returnValue;
+        if (JAVA_SOURCE_PARTTEN.matcher(change.getPath()).find()) {
+            final String fullName = convertJavaSourcePathToFullName(change.getPath());
+            if (fullName.endsWith(TEST_POSTFIX)) {
+                returnValue = NAME_TO_MODULE_EXTRAT_INFO.containsKey(
+                        fullName.substring(0, fullName.length() - TEST_POSTFIX.length()));
+            }
+            else {
+                returnValue = false;
+            }
+        }
+        else {
+            returnValue = false;
+        }
+        return returnValue;
+    }
+
+    /**
+     * Converts a {@link GitChange} of checkstyle module to a {@link ModuleExtractInfo}.
+     * @param change the change to convert
+     * @return the module info converted from the given change
+     */
+    public static ModuleExtractInfo convertModuleChangeToExtractInfo(GitChange change) {
+        final String fullName = convertJavaSourcePathToFullName(change.getPath());
+        return NAME_TO_MODULE_EXTRAT_INFO.get(fullName);
+    }
+
+    /**
+     * Converts a {@link GitChange} of test to the full name of corresponding checkstyle module.
+     * @param change the change to convert
+     * @return the full name of corresponding checkstyle module
+     */
+    public static String convertModuleTestChangeToModuleFullName(GitChange change) {
```

Comment:
These methods just seem too specific to be an actual utility method. 1 is unused, so it could be one issue for why I can't see a good way to fix it right now.

Yes, the map should stay.
I was thinking not to move them out but having simplier methods like `getModuleInfo` (take full module name as string parameter) so you don't have to move it out. The only issue for the test method would be `TEST_POSTFIX` which I don't see how you can get around for now.

Otherwise, you can throw the exception, but the test one also needs to make sure it ends with `Test`.
Usage of this method should never throw the exception, so you have to be real careful in it's use. I might make an issue to look at it later when we implement more in the module component.